### ISSUE TITLE
Does Autorest V3 support explode property in query?

### ISSUE THREAD

**Issue by priyankajayaswal1** [Azure/autorest#3604]
For our usecase, we have a query named ids which is array of strings with style form. with **explode=true** The url created in swagger ui looks correct as in of format: `https://editor.swagger.io/resource?ids=string&ids=string1&top=50` as opposed to `https://editor.swagger.io/Grower/ee/Boundary?ids=string,string1&top=50` ( -> this url & difference came with `explode` property use. In the generated client I see this happening:

`                _queryParameters.Add(string.Format("ids={0}", System.Uri.EscapeDataString(string.Join(",", ids))));
`

Do we not have this supported in autorest V3?

**Comment by lmazuel** [comment]
@pakrym ?

**Comment by pakrym** [comment]
As far as I know we don't

**Comment by lmazuel** [comment]
@pakrym explode support was added to m4 when we got the same issue for Python:
- https://github.com/Azure/autorest.modelerfour/issues/314
- https://github.com/Azure/autorest.python/issues/704
- https://github.com/Azure/autorest.testserver/pull/198

**Comment by daviwil** [comment]
Yep, we do have explode support.  Are you experiencing an issue with it @priyankajayaswal1?

**Comment by pakrym** [comment]
To clarify. I was only talking about the C# generator. M4 itself provides the right metadata in code model.


@lmazuel what's interesting is that we do pass the linked testserver test, I don't think it validates the explode though.

**Comment by priyankajayaswal1** [comment]
@daviwil yeah I am experiencing an issue. My bad should have mentioned C# in the description.
The issue is that I have a field named `ids` for which I'd want to have explode like behavior implies each `ids` attribute to translated to  one query param in url (like https://url?ids=a&ids=b) as opposed to https://url?ids=a,b,c.

- For this using expose solved the issue at swagger.json level (means in swagger ui by adding expose= true through our custom IOperationFilter, urls were getting properly reflected).
- But the issue is that when we try to use the client created with same swagger.json (specification 3.0.1), it appends `ids` query param as follows
`_queryParameters.Add(string.Format("ids={0}", System.Uri.EscapeDataString(string.Join(",", ids))));`

instead of 
```
       if (ids != null)
            {
                if (ids.Count == 0)
                {
                    _queryParameters.Add(string.Format("ids={0}", System.Uri.EscapeDataString(string.Empty)));
                }
                else
                {
                    foreach (var _item in ids)
                    {
                        _queryParameters.Add(string.Format("ids={0}", System.Uri.EscapeDataString("" + _item)));
                    }
                }
            }
```

 Note: ^ above is created when we use swagger spec2.0 instead.

In some of our projects, we do need to proceed with 3.0.1 and need the `ids` to work with expose functionality. So that becomes an issue since the client code isn't working as expected.

Also the command we are using is:
`autorest --csharp --v3 --input-file=$(ClientSwaggerFile) --output-folder=$(ClientOutDir) --clear-output-folder --verbose --namespace=$(ServicePrefix)Client.Autogenerated --version=3.0.6274`

Any suggestions if I am doing anything wrong?

**Comment by pakrym** [comment]
By running `autorest --csharp --v3` you are using V3 autorest pipeline with V2 C# generator. But that's not the biggest problem, neither V2 or V3 C# generator, currently, supports the explode options.

**Comment by priyankajayaswal1** [comment]
In swagger 2.0 explode equivalent is using below:
```
{
            "in": "query",
            "name": "parentIds",
            "description": "Parent Ids of the resource.",
            "type": "array",
            "items": {
              "type": "string"
            },
            "collectionFormat": "multi"
          },
```
which works fine with and without --v3 flag.

**Comment by Mikael-Gustavsson** [comment]
The bug in autorest is : **When explode is omitted for** `style": "form"`, **it defaults to false, which is not according to spec.**

My conclusion is that this is a bug in autorest, and a serious one at that since Autorest-generated APIs is not compatible with .net core actions handling array-based query parameters.
The Swagger specification on [query parameters](https://swagger.io/docs/specification/serialization/) states default behavior of array parameters to be:
` "style": "form",
            "explode": true,` ...and as such can, and may, be omitted,

To compound the problem: When using Swashbuckle to generate the parameter, it omits the `explode `variable when set to true.

This issue is related to https://github.com/Azure/autorest/issues/3519, and correct default handling should probably be addressed as one.

**Comment by Mikael-Gustavsson** [comment]
Ugly temporary workaround for **Swashbuckle** users having this problem:
Borrowing from https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/1701#issuecomment-643273567
You can amend the suggestion, linked above, to resolve the `explode: true` problem as well:
```C#
public class QueryArramParamFilter : IParameterFilter
{
	public void Apply(OpenApiParameter parameter, ParameterFilterContext context)
	{
		if (parameter.In.HasValue && parameter.In.Value == ParameterLocation.Query)
		{
			if (parameter.Schema?.Type == "array")
			{
				
				parameter.Style = ParameterStyle.Form;
				//parameter.Explode = true; //does not add the explode parameter to the swagger specification
				parameter.Extensions.Add("explode", new MyExplode());
			}
		}
	}
}

public class MyExplode : IOpenApiExtension
{
	public void Write(IOpenApiWriter writer, OpenApiSpecVersion specVersion)
	{
		writer.WriteRaw("true");
	}
}
```
... Above code will add the parameter `explode: true` to the swashbuckle generated output and will, when using autorest 3.0.6247, make the c# formatter in V3 mode generate desired parameter syntax for arrays
